lib/nx_signal/filters.ex

```diff
@@ -4,78 +4,92 @@ defmodule NxSignal.Filters do
   """
   import Nx.Defn
 
-  @pi :math.pi()
-
   @doc ~S"""
-  Calculates the normalized sinc function $sinc(t) = \frac{sin(\pi t)}{\pi t}$
+  Performs a median filter on a rank 1 or rank 2 tensor.
 
-  ## Examples
+  ## Options
 
-      iex> NxSignal.Filters.sinc(Nx.tensor([0, 0.25, 1]))
-      #Nx.Tensor<
-        f32[3]
-        [1.0, 0.9003162980079651, -2.7827534054836178e-8]
-      >
+    * `:kernel_shape` - the shape of the sliding window.
+    It must be compatible with the shape of the tensor.
   """
   @doc type: :filters
-  defn sinc(t) do
-    t = t * @pi
-    zero_idx = Nx.equal(t, 0)
+  def median(t = %Nx.Tensor{shape: {length}}, opts) do
```

Comment:
```suggestion
  deftransform median(t = %Nx.Tensor{shape: {length}}, opts) do
```

Use this on the other clause to. Otherwise, this won't be usable from within `defn`

Also, I believe that you'll need to reshape the result back into a 1D tensor

Reply:
got it. I did reshape the result by squeezing the result on the `median_n` output. Should I move that to the 1D case only?

Reply:
Problem is if the input is `{1, x}`, then the output will be `{1, x}` and you'll end up squeezing it into 1D when you shouldn't